### PR TITLE
changes to enable ocn->glc coupling at multiple leves

### PR DESCRIPTION
**Pull request by mvertens**
### Description of changes
Incorporate changes  to enable ocn->glc coupling at multiple levels.

### Specific notes
This PR uses dynamic masking in the prep_phases_glc_mod in order to enable sending ocn data to glc. The ocean data is at multiple levels but with associated differences in the ocn bathymetry.

Contributors other than yourself, if any: 

CMEPS Issues Fixed (include github issue #):

Are changes expected to change answers?
 - [x] bit for bit
 - [ ] different at roundoff level
 - [ ] more substantial 

Any User Interface Changes (namelist or namelist defaults changes)?
 - [x] Yes - added new nuopc_config variables ocn2glc_coupling and ocn2glc_levels (colon delimited string of ocean levels)
 - [ ] No

Testing performed if application target is CESM:(either UFS-S2S or CESM testing is required):
- [ ] (recommended) CIME_DRIVER=nuopc scripts_regression_tests.py
   - machines:
   - details (e.g. failed tests):
- [x] (recommended) CESM testlist_drv.xml
   - machines and compilers: cheyenne/intel
   - details (e.g. failed tests): compared to cmeps baselines dec06 and created new baseline dec19
- [ ] (optional) CESM prealpha test
   - machines and compilers
   - details (e.g. failed tests):
- [ ] (other) please described in detail
   - machines and compilers
   - details (e.g. failed tests):

Testing performed if application target is UFS-S2S:
- [ ] (recommended) UFS-S2S testing
   - description:
   - details (e.g. failed tests):

Testing performed if application target is UFS-HAFS:
- [ ] (recommended) UFS-HAFS testing
   - description:
   - details (e.g. failed tests):

Hashes used for testing:
- [x] CESM:
  - repository to check out: https://github.com/ESCOMP/CESM.git
  - branch: nuopc_dev_development
  - hash: 1c90e71
- [ ] UFS-S2S, then umbrella repostiory to check out and associated hash:
  - repository to check out:
  - branch:
  - hash:
- [ ] UFS-HAFS, then umbrella repostiory to check out and associated hash:
  - repository to check out:
  - branch:
  - hash:

### CODE REVIEW

File: cime_config/namelist_definition_drv.xml [hunk 1]
```diff
@@ -768,6 +768,31 @@
     </values>
   </entry>
 
+  <entry id="ocn2glc_coupling">
+    <type>logical</type>
+    <category>flds</category>
+    <group>ALLCOMP_attributes</group>
+    <desc>
+      .true. if ocean is sends fields at multiple ocean levels to the land-ice component
```
Comment (jedwards4b) [inline cime_config/namelist_definition_drv.xml:776]: Please consider wording this differently.

File: cime_config/namelist_definition_drv.xml [hunk 2]
```diff
@@ -768,6 +768,31 @@
...
+      <value>.false.</value>
+    </values>
+  </entry>
+
+  <entry id="ocn2glc_levels">
+    <type>char</type>
+    <category>flds</category>
+    <group>ALLCOMP_attributes</group>
+    <desc>
+      if ocean is sends fields at multiple ocean levels to the
```
Comment (jedwards4b) [inline cime_config/namelist_definition_drv.xml:788]: desc needs better wording

Comment (jedwards4b) [reply cime_config/namelist_definition_drv.xml:788]: Why would ocn send to glc at multiple levels?  I don't understand.

**Review by jedwards4b — APPROVED** [review]
minor nits about comments

File: cime_config/namelist_definition_drv.xml (hunk 2 above)
Comment (mvertens) [reply cime_config/namelist_definition_drv.xml:788]: I'm not sure how to rephrase this. The ocean is sending internal temperature and salinity at 10 levels to the glc. Each level will have a different bathymetry. CISM needs this when it couples the antarctic ice shelf to the ocean. So what would you like me to say? If you can suggest any different wording I'm happy to put it in.

Comment (jedwards4b) [reply cime_config/namelist_definition_drv.xml:788]: "if ocean is sends" maybe just remove the word "is"?

**Comment by DeniseWorthen** [comment]
I've tested in my fork [#b3c83f7](https://github.com/DeniseWorthen/ufs-weather-model/tree/feature/updcmeps) and all ufs-cpld and ufs-datm test pass.

**Review by DeniseWorthen — APPROVED** [review]

**Pull request merged**